packages/sdk/src/server/index.ts

```diff
@@ -404,64 +450,86 @@ class ServerClient {
       "POST",
       "PUT",
       "PATCH",
-    ].includes(method.toUpperCase()) && body) {
-      requestOptions.body = body;
+    ].includes(method.toUpperCase()) && processedBody) {
+      requestOptions.body = processedBody;
     }
 
     const response: Response = await fetch(url.toString(), requestOptions);
 
     if (!response.ok) {
-      throw new Error(`HTTP error! status: ${response.status}`);
+      const errorBody = await response.text();
+      throw new Error(`HTTP error! status: ${response.status}, body: ${errorBody}`);
     }
 
-    const result = await response.json() as unknown as T;
-    return result;
+    // Attempt to parse JSON, fall back to raw text if it fails
+    const contentType = response.headers.get("Content-Type");
+    if (contentType && contentType.includes("application/json")) {
+      return await response.json() as T;
+    }
+
+    return await response.text() as unknown as T;
   }
 
   /**
-   * Makes a request to the Pipedream API.
+   * Makes a request to the Pipedream API with appropriate authorization.
    *
    * @template T - The expected response type.
    * @param path - The API endpoint path.
    * @param opts - The options for the request.
+   * @param authType - The type of authorization to use ("oauth" or "connect").
    * @returns A promise resolving to the API response.
    * @throws Will throw an error if the response status is not OK.
    */
-  async _makeApiRequest<T>(
+  private async makeAuthorizedRequest<T>(
     path: string,
     opts: RequestOptions = {},
+    authType: "oauth" | "connect" = "oauth",
   ): Promise<T> {
-    const headers = {
-      ...opts.headers ?? {},
-      "Authorization": await this._oauthAuthorizationHeader(),
+    const headers: Record<string, string> = {
+      "Content-Type": "application/json",
+      ...opts.headers,
     };
-    return this._makeRequest<T>(path, {
+
+    if (authType === "oauth") {
+      headers["Authorization"] = await this.oauthAuthorizationHeader();
+    } else {
+      headers["Authorization"] = this.connectAuthorizationHeader();
+    }
```

Comment:
I'm now more convinced that we should split these clients into `ConnectClient` and `ApiClient`, WDYT?
We could also rename `ServerClient` (too verbose IMHO) to just `Client`, make it abstract, and implement the HTTP calling logic only.

```typescript
abstract class Client {
  private baseUrl: string;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  makeRequest(
    method: "GET" | "POST",
    path: string,
    headers: object,
    payload?: unknown,
  ) {
    // Do all the stuff that it does now
  }
}

class ConnectClient extends Client {
  private publicKey: string;
  private secretKey: string;

  constructor(publicKey: string, secretKey: string) {
    super("https://api.pipedream.com/v1/connect");
    this.publicKey = publicKey;
    this.privateKey = privateKey;
  }

  private makeRequest(
    method: "GET" | "POST",
    path: string,
    headers: object,
    payload?: unknown,
  ) {
    headers["Authorization"] = this.authorizationHeader();
    return super(method, path, headers, payload);
  }
}
```

Reply:
Not now of course.

Reply:
100%, maybe after we ship all the JWT validation in catcher and Rails we can work on this next sprint